File: python/Quesn10.py
# Name : Himanshu
# Roll No. : CS20B1097

'''
Question 10. 
Write a program that accepts a sentence and calculate the number of letters and digits.
'''

sentence = input()
letter = digit = 0

for character in sentence:
    if character.isdigit():
        digit = digit+1
    elif character.isalpha():
        letter = letter+1
    else:
        pass

print(letter, digit)

File: cpp/quesn10.cpp
/* Write a program that accepts a sentence and calculate the number of letters and digits. */
#include <iostream>
#include <string>
using namespace std;

int letterNumber(string input){
    int result = 0;
    for(int i = 0; i < input.size(); i++){
        if((input[i]>=65 && input[i]<=90)|| (input[i]>=97 && input[i]<=122) )
            result++;
    }
    return result;
}

int digitNumber(string input){
    int result = 0;
    for(int i = 0; i < input.size(); i++){
        if(input[i]>=48 && input[i]<=57)
            result++;
    }
    return result;
}


int main(){
    string sentence;
    cout<<"Please enter a sentence: ";
    getline(cin >> ws, sentence);
    cout <<"The given sentence is " << sentence << endl;
    cout <<"Number of Letters: " << letterNumber(sentence) << endl;
    cout <<"Number of Digits: " << digitNumber(sentence) << endl;
    return 0;
}